python/ray/serve/tests/test_deploy.py

```diff
@@ -123,6 +124,22 @@ def __call__(self, *args):
     assert val5 == "4"
 
 
+def test_deploy_handle_validation(serve_instance):
```

Comment:
please add some tests for edge cases like when the handle is not registered yet and when the user calls a method that doesn't exist.